pkg/resourcequotas/resourcequotas.go

```diff
@@ -0,0 +1,198 @@
+)
+
+// Builder provides struct for resource quotas containing connection to the cluster.
+type Builder struct {
+	// Resource Quota definition
+	Definition *corev1.ResourceQuota
+
+	// Created resource quota object
+	Object *corev1.ResourceQuota
+
+	// Used in functions that define or mutate deployment definition. errorMsg is processed before the deployment
+	// object is created.
+	errorMsg  string
+	apiClient corev1Typed.CoreV1Interface
+}
+
+// NewBuilder creates a new resource quota builder.
+func NewBuilder(apiClient corev1Typed.CoreV1Interface, name, nsname string,
+	quotaDefinition corev1.ResourceQuota) *Builder {
+	glog.V(100).Infof("Initializing new resource quota structure with the following params: "+
+		"name=%s, namespace=%s, quotaSpec=%v", name, nsname, quotaDefinition)
+
+	builder := &Builder{
+		Definition: &quotaDefinition,
+		apiClient:  apiClient,
+	}
+
+	if name == "" {
+		glog.V(100).Info("Resource Quota name is empty")
+
+		builder.errorMsg = "resource quota 'name' cannot be empty"
+
+		return builder
+	}
+
+	if nsname == "" {
+		glog.V(100).Info("Resource Quota namespace is empty")
+
+		builder.errorMsg = "resource quota 'namespace' cannot be empty"
+
+		return builder
+	}
+
+	return builder
+}
+
+// Pull retrieves the resource quota from the cluster.
+func Pull(apiClient *clients.Settings, name, nsname string) (*Builder, error) {
+	if apiClient == nil {
+		glog.V(100).Info("apiClient is nil")
+
+		return nil, fmt.Errorf("apiClient is nil")
+	}
+
+	glog.V(100).Infof("Pulling resource quota with the following params: name=%s, namespace=%s", name, nsname)
+
+	builder := &Builder{
+		apiClient: apiClient.CoreV1Interface,
+		Definition: &corev1.ResourceQuota{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Info("Resource Quota name is empty")
+
+		return nil, fmt.Errorf("resource quota 'name' cannot be empty")
+	}
+
+	if nsname == "" {
+		glog.V(100).Info("Resource Quota namespace is empty")
+
+		return nil, fmt.Errorf("resource quota 'namespace' cannot be empty")
+	}
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("resource quota %s does not exist in namespace %s",
+			name, nsname)
+	}
+
+	builder.Definition = builder.Object
+
+	return builder, nil
+}
+
+// Exists checks if the resource quota exists in the cluster.
+func (builder *Builder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if resource quota %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+
+	builder.Object, err = builder.apiClient.ResourceQuotas(
+		builder.Definition.Namespace).Get(context.TODO(),
+		builder.Definition.Name, metav1.GetOptions{})
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// validate validates the resource quota definition.
+func (builder *Builder) validate() (bool, error) {
```

Comment:
unexported validate function should go after the exported functions

Reply:
Done.

Reply:
@sebrandon1 It doesn't look like this change is in the PR, does it still need to be pushed?